spec/models/post_spec.rb

```diff
@@ -1,5 +1,45 @@
 require 'rails_helper'
 
 RSpec.describe Post, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  user = User.new(name: 'Saied', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'developer')
+  subject { Post.new(author: user, title: 'title', text: 'text') }
+  before { subject }
+
+  it 'author is nil' do
+    test_case = subject
+    test_case.author = nil
+    expect(test_case).to_not be_valid
+  end
+  it 'title is nil or blank or less than length of 4' do
+    test_case = subject
+    test_case.title = nil
+    expect(test_case).to_not be_valid
+    test_case.title = ''
+    expect(test_case).to_not be_valid
+  end
+  it 'title is less than length of 4' do
+    test_case = subject
+    test_case.title = 'sai'
+    expect(test_case).to_not be_valid
+  end
+  it 'title is longer than length of 250' do
+    test_case = subject
+    test_case.title = 'a' * 251
+    expect(test_case).to_not be_valid
+  end
+  it 'title is not string' do
+    test_case = subject
+    test_case.title = 5
+    expect(test_case).to_not be_valid
+  end
+  it 'comments counter is not number' do
+    test_case = subject
+    test_case.comments_counter = 'hi'
+    expect(test_case).to_not be_valid
+  end
+  it 'likes counter is not number' do
+    test_case = subject
+    test_case.likes_counter = 'hi'
+    expect(test_case).to_not be_valid
+  end
```

Comment:
- [ ] You did an excellent job with the unit tests for the User, Post, Like, and Comment models. However, it would be better to include some tests that expect objects to be valid in order to confirm that the model validations have been well tested.

Reply:
@medaminedev66  Please check it out again now fixed

Reply:
10min left